tools/snapshot_bot_remote_display/snapshot_bot_remote_display.cc

```diff
@@ -0,0 +1,436 @@
+                }
+
+                // Update best snapshot
+                if(m_BestSnapshotNetSource.readFrame(m_ReceiveBuffer)) {
+                    if(m_BestSnapshotReceived) {
+                        LOGW << "Best snapshot receiving out of sync";
+                    }
+                    else {
+                        // Make a copy of received image
+                        m_ReceiveBuffer.copyTo(m_BestSnapshot);
+
+                        // Set flag
+                        m_BestSnapshotReceived = true;
+                    }
+                }
+
+                // Update snapshot that robot matched with best snapshot
+                if(m_SnapshotNetSource.readFrame(m_ReceiveBuffer)) {
+                    if(m_SnapshotReceived) {
+                        LOGW << "Snapshot receiving out of sync";
+                    }
+                    else {
+                        // Make a copy of received image
+                        m_ReceiveBuffer.copyTo(m_Snapshot);
+
+                        // Set flag
+                        m_SnapshotReceived = true;
+                    }
+                }
+
+                // If both images have been received
+                if(m_BestSnapshotReceived && m_SnapshotReceived) {
+                    BOB_ASSERT(m_Snapshot.type() == CV_8UC1);
+
+                    const int imageSize = m_Snapshot.size().width * m_Snapshot.size().height;
+                    const int imageWidth = m_Snapshot.size().width;
+                    const int numScanColumns = (int)std::round(units::angle::turn_t(m_Config.getMaxSnapshotRotateAngle()).value() * (double)imageWidth);
+
+                    // Ensure there is enough space in RIDF
+                    m_RIDF.resize(numScanColumns * 2);
+
+                    // Create functor to calculate difference between images
+                    Navigation::AbsDiff differencer(imageWidth);
+
+                    // Scan across columns on left of image
+                    cv::Mat mask;
+                    auto rotatorLeft = Navigation::InSilicoRotater::create(m_Snapshot.size(), mask, m_Snapshot,
+                                                                           1, 0, numScanColumns);
+                    rotatorLeft.rotate(
+                            [imageSize, numScanColumns, &differencer, this]
+                            (const cv::Mat &fr, const cv::Mat &, size_t i)
+                            {
+                                // Calculate image difference
+                                auto diffIter = differencer(fr, m_BestSnapshot, m_SnapshotDifferenceScratch);
+
+                                float sumDifference = std::accumulate(diffIter, diffIter + imageSize, 0.0f);
+
+                                // Store mean difference in RIDF
+                                m_RIDF[numScanColumns - 1 - i] = Navigation::AbsDiff::mean(sumDifference, imageSize);
+                            });
```

Comment:
Why are you filling ``m_RIDF`` from the end here?

Reply:
Ahh I see. nvm.

An alternative to having ``rotatorLeft`` and ``rotatorRight`` would be to just have a single ``rotator`` then doing ``std::rotate`` on it. But this is fine too

Reply:
...although, on third thoughts, I think I might not understand this after all :wink:. Is it just that you want the RIDF centred on 0°? If so, you still wouldn't want ``m_RIDF`` reverse-filled, just offset... right?

Reply:
So we're only rotating +-90 degrees and that gives you two small RIDFs, one of which is reversed. This code combines them together into an RIDF centred at 0 degrees as you say.